sdcm/sla/sla_tests.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python
+import logging
+import time
+import traceback
+
+from sdcm.sct_events import Severity
+from sdcm.sct_events.workload_prioritization import WorkloadPrioritizationTestEvent
+from sdcm.sla.libs.sla_utils import SlaUtils
+from sdcm.utils.loader_utils import DEFAULT_USER, DEFAULT_USER_PASSWORD
+from test_lib.sla import create_sla_auth
+
+LOGGER = logging.getLogger(__name__)
+
+
+class Steps(SlaUtils):
+    # pylint: disable=too-many-arguments
+    def run_stress_and_validate_scheduler_runtime_during_load(self, tester, read_cmds, prometheus_stats, read_users,
+                                                              stress_queue, sleep=600):
+        # pylint: disable=not-context-manager
+        with WorkloadPrioritizationTestEvent(
+                step="Run stress command and validate scheduler runtime during load") as wp_event:
+            try:
+                start_time = time.time() + 60
+                # pylint: disable=protected-access
+                tester._run_all_stress_cmds(stress_queue, params={'stress_cmd': read_cmds, 'round_robin': True})
+                time.sleep(sleep)
+                self.validate_if_scylla_load_high_enough(start_time=start_time,
+                                                         wait_cpu_utilization=self.MIN_CPU_UTILIZATION,
+                                                         prometheus_stats=prometheus_stats)
+                end_time = time.time()
+
+                runtime_per_role_before_increase = self.validate_scheduler_runtime(start_time=start_time,
+                                                                                   end_time=end_time,
+                                                                                   read_users=read_users,
+                                                                                   prometheus_stats=prometheus_stats,
+                                                                                   db_cluster=tester.db_cluster)
+                # TODO: add validation
+                LOGGER.debug('RUN TIME PER ROLE BEFORE INCREASE: {}'.format(runtime_per_role_before_increase))
+            except Exception as details:  # pylint: disable=broad-except
+                wp_event.add_error([str(details)])
```

Comment:
maybe we can return the event object, and collect the steps failures, and raise a failure with full_traceback is there is one.

Reply:
done